### PR TITLE
feature/SDK-39_well-known-oidf

### PR DESCRIPTION
**Pull request by sanderPostma**

### CODE REVIEW

File: packages/callback-example/lib/__tests__/issuerCallback.spec.ts [hunk 1]
```diff
@@ -24,6 +24,9 @@ import { DIDDocument } from 'did-resolver'
 import * as jose from 'jose'
 
 import { generateDid, getIssuerCallbackV1_0_11, getIssuerCallbackV1_0_13, verifyCredential } from '../IssuerCallback'
+import {
+  AuthorizationServerMetadataBuilder
+} from '@sphereon/oid4vci-issuer/dist/builder/AuthorizationServerMetadataBuilder'
```
Comment (nklomp) [inline packages/callback-example/lib/__tests__/issuerCallback.spec.ts:29]: Incorrect import

File: packages/issuer-rest/lib/__tests__/ClientIssuerIT.spec.ts [hunk 2]
```diff
@@ -13,10 +13,13 @@ import {
...
   OpenId4VCIVersion,
   PRE_AUTH_CODE_LITERAL,
-  PRE_AUTH_GRANT_LITERAL,
+  PRE_AUTH_GRANT_LITERAL
 } from '@sphereon/oid4vci-common'
 import { VcIssuer } from '@sphereon/oid4vci-issuer/dist/VcIssuer'
 import { CredentialSupportedBuilderV1_13, VcIssuerBuilder } from '@sphereon/oid4vci-issuer/dist/builder'
+import {
+  AuthorizationServerMetadataBuilder
+} from '@sphereon/oid4vci-issuer/dist/builder/AuthorizationServerMetadataBuilder'
```
Comment (nklomp) [inline packages/issuer-rest/lib/__tests__/ClientIssuerIT.spec.ts:22]: Import error

File: packages/issuer-rest/lib/__tests__/IssuerTokenServer.spec.ts [hunk 3]
```diff
@@ -12,6 +12,9 @@ import {
   URIState,
 } from '@sphereon/oid4vci-common'
 import { VcIssuer } from '@sphereon/oid4vci-issuer'
+import {
+  AuthorizationServerMetadataBuilder
+} from '@sphereon/oid4vci-issuer/dist/builder/AuthorizationServerMetadataBuilder'
```
Comment (nklomp) [inline packages/issuer-rest/lib/__tests__/IssuerTokenServer.spec.ts:17]: Import error.

File: packages/issuer-rest/lib/oid4vci-api-functions.ts [hunk 4]
```diff
@@ -392,11 +394,24 @@ export function pushedAuthorizationEndpoint<DIDDoc extends object>(
...
-  })
+  }
+  router.get(WellKnownEndpoints.OPENID4VCI_ISSUER, credentialIssuerHandler)
+
+  const authorizationServerHandler = (request: Request, response: Response) => {
+    return response.send(issuer.authorizationServerMetadata)
+  }
+  router.get(WellKnownEndpoints.OAUTH_AS, authorizationServerHandler)
+  
+  const openidFederationHandler = (request: Request, response: Response) => {
```
Comment (nklomp) [inline packages/issuer-rest/lib/oid4vci-api-functions.ts:408]: Why is a REST API implementation of a OID4VC protocol which has little relation to OIDF all of a sudden handling OIDF metadata and wellknowns?

File: packages/issuer/lib/VcIssuer.ts [hunk 5]
```diff
@@ -33,10 +33,11 @@ import {
   toUniformCredentialOfferRequest,
   TxCode,
   TYP_ERROR,
-  URIState,
+  URIState
 } from '@sphereon/oid4vci-common'
 import { CredentialEventNames, CredentialOfferEventNames, EVENTS } from '@sphereon/oid4vci-common'
 import { CredentialIssuerMetadataOptsV1_0_13 } from '@sphereon/oid4vci-common'
+import { OpenidFederationMetadata } from '@sphereon/oid4vci-common/dist/types/OpenidFederationMetadata'
```
Comment (nklomp) [inline packages/issuer/lib/VcIssuer.ts:40]: Please be more careful with your imports. There are multiple occasions I have to point out

File: packages/issuer/lib/VcIssuer.ts [hunk 6]
```diff
@@ -45,6 +46,8 @@ import { CredentialDataSupplier, CredentialDataSupplierArgs, CredentialIssuanceI
 
 export class VcIssuer<DIDDoc extends object> {
   private readonly _issuerMetadata: CredentialIssuerMetadataOptsV1_0_13
+  private readonly _authorizationServerMetadata: AuthorizationServerMetadata
+  private readonly _openidFederationMetadata?: OpenidFederationMetadata
```
Comment (nklomp) [inline packages/issuer/lib/VcIssuer.ts:50]: Should not be here

File: packages/issuer/lib/builder/VcIssuerBuilder.ts [hunk 7]
```diff
@@ -9,9 +11,9 @@ import {
   MetadataDisplay,
   TokenErrorResponse,
   TxCode,
-  URIState,
+  URIState
 } from '@sphereon/oid4vci-common'
-import { CredentialIssuerMetadataOptsV1_0_13 } from '@sphereon/oid4vci-common'
+import { OpenidFederationMetadata } from '@sphereon/oid4vci-common/dist/types/OpenidFederationMetadata'
```
Comment (nklomp) [inline packages/issuer/lib/builder/VcIssuerBuilder.ts:16]: Imports again!

File: packages/issuer/lib/builder/VcIssuerBuilder.ts [hunk 8]
```diff
@@ -22,6 +24,8 @@ import { IssuerMetadataBuilderV1_13 } from './IssuerMetadataBuilderV1_13'
 export class VcIssuerBuilder<DIDDoc extends object> {
   issuerMetadataBuilder?: IssuerMetadataBuilderV1_13
   issuerMetadata: Partial<CredentialIssuerMetadataOptsV1_0_13> = {}
+  authorizationServerMetadata: Partial<AuthorizationServerMetadata> = {}
+  openidFederationMetadata: Partial<OpenidFederationMetadata> = {}
```
Comment (nklomp) [inline packages/issuer/lib/builder/VcIssuerBuilder.ts:28]: Should not be here

**Review by nklomp — CHANGES_REQUESTED** [review]
1 Please be more careful with your imports. Mulltiple of them are off
2 OIDF has nothing todo with OID4VC, except for the fact it can be used as a registry for trust establishment. So it for sure does not belong in the VCI REST API or anything

**Comment by sanderPostma** [comment]
I wasn't done yet and also I did not review myself yet. I created a draft PR in IJ, but apparently it created a regular one.

File: packages/callback-example/lib/__tests__/issuerCallback.spec.ts (hunk 1 above)
Comment (sanderPostma) [reply packages/callback-example/lib/__tests__/issuerCallback.spec.ts]: fixed

File: packages/issuer-rest/lib/__tests__/ClientIssuerIT.spec.ts (hunk 2 above)
Comment (sanderPostma) [reply packages/issuer-rest/lib/__tests__/ClientIssuerIT.spec.ts]: fixed

File: packages/issuer-rest/lib/__tests__/IssuerTokenServer.spec.ts (hunk 3 above)
Comment (sanderPostma) [reply packages/issuer-rest/lib/__tests__/IssuerTokenServer.spec.ts]: fixed

File: packages/issuer/lib/VcIssuer.ts (hunk 5 above)
Comment (sanderPostma) [reply packages/issuer/lib/VcIssuer.ts]: fixed, but going to be removed

File: packages/issuer/lib/builder/VcIssuerBuilder.ts (hunk 7 above)
Comment (sanderPostma) [reply packages/issuer/lib/builder/VcIssuerBuilder.ts]: fixed, but going to be removed

**Comment by sanderPostma** [comment]
Closing as the code has nothing to do with this PR anymore

**Pull request closed without merging**